crates/oxc_parser/src/js/statement.rs

```diff
@@ -269,6 +277,32 @@ impl<'a> Parser<'a> {
             return self.parse_for_loop(span, init, r#await);
         }
 
+        if (self.cur_kind() == Kind::Await && self.peek_kind() == Kind::Using)
+            || (self.cur_kind() == Kind::Using && self.peek_kind() == Kind::Ident)
+        {
+            let using_decl = self.parse_using_declaration(StatementContext::For)?;
+
+            if matches!(self.cur_kind(), Kind::In) {
+                if using_decl.is_await {
+                    self.error(diagnostics::AwaitUsingDeclarationNotAllowedInForInStatement(
+                        using_decl.span,
+                    ));
+                } else {
+                    self.error(diagnostics::UsingDeclarationNotAllowedInForInStatement(
+                        using_decl.span,
+                    ));
+                }
+            }
+
+            if matches!(self.cur_kind(), Kind::In | Kind::Of) {
+                let init = ForStatementLeft::UsingDeclaration(self.ast.alloc(using_decl));
+                return self.parse_for_in_or_of_loop(span, r#await, init);
+            }
+
+            let init = Some(ForStatementInit::UsingDeclaration(self.ast.alloc(using_decl)));
+            return self.parse_for_loop(span, init, r#await);
+        }
```

Comment:
nit: refactor this into a function, maybe check how typescript parses this?

Reply:
like extract the logic for var decl/ using decl into a function?

Reply:
thanks for clarifying, will update

Reply:
Mind taking a look. I wanted to extrac the two `parse_variable_declaration_for_statement` and `parse_using_declaration_for_statement` but I could't find a good way to do it